### PR TITLE
Add ubuntu-frame and glmark2 tests

### PR DESCRIPTION
**Pull request by patliuu**
This PR introduced tests for `ubuntu-frame` and `glmark2-es2-wayland` benchmark.

For ubuntu-frame, we checked whether it is already launching in the background
* If yes, we dump the journal log relevant to ubuntu-frame
* If not, we try to launch it for 20 seconds to see if there's any launching error.

For glmark2-es2-wayland benchmark, we run it by `graphics-test-tools`
* Try to see whether we can run the benchmark successfully and get the glmark2 score.
* Check whether the `GL_VENDOR` and `GL_RENDERER` is correct.
  * To make sure the benchmark is running on the GPU, not by CPU software rendering (which the renderer starts with `llvm`)
  * `GL_VENDOR` and `GL_RENDERER` need to be specified in the config variables.

Also added an ubuntu-frame manifest.

Sideload test results:
* On Shiner MB touchscreen (uc20)
https://pastebin.canonical.com/p/6TX9dQF3Tb/
* On Mering device (uc22)
https://pastebin.canonical.com/p/SCRSBhcsjG/
* On Baoshan device (uc22) (both ubuntu-frame active and inactive before running the test)
https://pastebin.canonical.com/p/nkWwHtX4Ns/

### CODE REVIEW

File: checkbox-provider-ce-oem/units/mir/jobs.pxu [hunk 1]
```diff
@@ -0,0 +1,29 @@
+id: mir/check-ubuntu-frame-launching-auto
+category_id: mir_test
+plugin: shell
+flags: also-after-suspend
+user: root
+imports:
+    from com.canonical.certification import snap
```
Comment (seankingyang) [inline checkbox-provider-ce-oem/units/mir/jobs.pxu:7]: I am curious about this line is necessary or not, if we want to use the `sanp` in `requires` field ~~
Cos i still cna use the `snap.name` w/o add that ~~

**Review by seankingyang — APPROVED** [review]
It looks great, and simplify some shell cmd part.
Everything work well ~~

File: checkbox-provider-ce-oem/units/mir/jobs.pxu (hunk 1 above)
Comment (patliuu) [reply checkbox-provider-ce-oem/units/mir/jobs.pxu:7]: Not sure what was your environment. I just tested it again, without `from com.canonical.certification import snap` the job will not be generated.
```
# with importing snap
ceqa@ubuntu:~$checkbox.checkbox-cli list-bootstrapped com.canonical.qa.ceoem::ce-oem-mir-automated
Using sideloaded provider: checkbox-provider-ce-oem, version 0.1 from /var/tmp/checkbox-providers/checkbox-provider-ce-oem
Skipped file: /var/tmp/checkbox-providers/checkbox-provider-ce-oem/bin/socketcan_busoff_test.py
com.canonical.certification::snap
com.canonical.qa.ceoem::mir/check-ubuntu-frame-launching-auto
com.canonical.qa.ceoem::mir/glmark2-es2-wayland-auto
ceqa@ubuntu:~$

# without importing snap
ceqa@ubuntu:~$ checkbox.checkbox-cli list-bootstrapped com.canonical.qa.ceoem::ce-oem-mir-automated
Using sideloaded provider: checkbox-provider-ce-oem, version 0.1 from /var/tmp/checkbox-providers/checkbox-provider-ce-oem
Skipped file: /var/tmp/checkbox-providers/checkbox-provider-ce-oem/bin/socketcan_busoff_test.py
ceqa@ubuntu:~$

File: checkbox-provider-ce-oem/bin/graphics_test.sh [hunk 2]
```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+test_ubuntu_frame_launching() {
+    if pgrep -if "ubuntu-frame" > /dev/null; then
+        echo "The ubuntu-frame is active"
+        echo "No need to bring it up again"
+        echo
+        echo "journal log of ubuntu frame:"
+        journalctl -b 0 | grep "ubuntu-frame"
+        touch "$PLAINBOX_SESSION_SHARE/ubuntu-frame-is-activating"
```
Comment (stanley31huang) [inline checkbox-provider-ce-oem/bin/graphics_test.sh:10]: we could simply grep journal log by `journalctl -b 0 -g "ubuntu-frame".
another issue is are we going to dump the journal log into 'ubuntu-frame-is-activating'?

File: checkbox-provider-ce-oem/bin/graphics_test.sh [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        timeout 20s ubuntu-frame || ( [[ $? -eq 124 ]] && \
+        echo -e "\nPASS: Timeout reached without any failures detected." )
+    fi
+}
+
+test_glmark2_es2_wayland() {
+    CMD="env XDG_RUNTIME_DIR=/run/user/0 graphics-test-tools.glmark2-es2-wayland"
+    OUTPUT=""
+    EXIT=0
+    if [ -f "$PLAINBOX_SESSION_SHARE/ubuntu-frame-is-activating" ]; then
```
Comment (stanley31huang) [inline checkbox-provider-ce-oem/bin/graphics_test.sh:27]: I don't think we need the 'ubuntu-frame-is-activating' file to check the ubuntu-frame is active or not.

**Review by stanley31huang — APPROVED** [review]
LGTM

**Pull request merged**